google-cloud-spanner/src/test/java/com/google/cloud/spanner/DatabaseClientImplTest.java

```diff
@@ -2381,4 +2393,565 @@ public void testAnalyzeUpdateStatement() {
+        int col = 0;
+        assertAsString("true", resultSet, col++);
+        assertAsString("100", resultSet, col++);
+        assertAsString("3.14", resultSet, col++);
+        assertAsString("6.626", resultSet, col++);
+        assertAsString("test-string", resultSet, col++);
+        assertAsString("{\"key1\": \"value1\"}", resultSet, col++);
+        assertAsString(
+            Base64.getEncoder().encodeToString("test-bytes".getBytes(StandardCharsets.UTF_8)),
+            resultSet,
+            col++);
+        assertAsString("2023-01-11", resultSet, col++);
+        assertAsString("2023-01-11T11:55:18.123456789Z", resultSet, col++);
+
+        assertAsString(ImmutableList.of("true", "NULL", "false"), resultSet, col++);
+        assertAsString(
+            ImmutableList.of(
+                String.format("%d", Long.MAX_VALUE), String.format("%d", Long.MIN_VALUE), "NULL"),
+            resultSet,
+            col++);
+        assertAsString(ImmutableList.of("NULL", "-12345.6789", "3.14"), resultSet, col++);
+        assertAsString(ImmutableList.of("6.626", "NULL", "-8.9123"), resultSet, col++);
+        assertAsString(ImmutableList.of("test-string1", "NULL", "test-string2"), resultSet, col++);
+        assertAsString(
+            ImmutableList.of("{\"key\": \"value1\"}", "{\"key\": \"value2\"}", "NULL"),
+            resultSet,
+            col++);
+        assertAsString(
+            ImmutableList.of(
+                String.format(
+                    "%s",
+                    Base64.getEncoder()
+                        .encodeToString("test-bytes1".getBytes(StandardCharsets.UTF_8))),
+                String.format(
+                    "%s",
+                    Base64.getEncoder()
+                        .encodeToString("test-bytes2".getBytes(StandardCharsets.UTF_8))),
+                "NULL"),
+            resultSet,
+            col++);
+        assertAsString(ImmutableList.of("2000-02-29", "NULL", "2000-01-01"), resultSet, col++);
+        assertAsString(
+            ImmutableList.of("2023-01-11T11:55:18.123456789Z", "NULL", "2023-01-12T11:55:18Z"),
+            resultSet,
+            col++);
+
+        assertFalse(resultSet.next());
+      }
+    }
+  }
+
+  @Test
+  public void testSelectUnknownType() {
+    mockSpanner.putStatementResult(
+        StatementResult.query(
+            Statement.of("SELECT * FROM foo"),
+            com.google.spanner.v1.ResultSet.newBuilder()
+                .setMetadata(
+                    ResultSetMetadata.newBuilder()
+                        .setRowType(
+                            StructType.newBuilder()
+                                .addFields(
+                                    Field.newBuilder()
+                                        .setName("c")
+                                        .setType(
+                                            Type.newBuilder()
+                                                .setCodeValue(Integer.MAX_VALUE)
```

Comment:
This line sets a type code value that does not exist, and will be seen as an unrecognized type in the client library